game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -1202,6 +1162,86 @@ && canAirSafelyLandAfterAttack(unit, t)) {
     }
   }
 
+  private void determineTerritoriesThatCanBeBombed(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Unit> alreadyAttackedWithUnits) {
+    final Map<Unit, Set<Territory>> bomberMoveMap =
+        territoryManager.getAttackOptions().getBomberMoveMap();
+    final Set<Unit> bombers = bomberMoveMap.keySet();
+    bombers.removeAll(alreadyAttackedWithUnits);
+    final boolean raidsMayBePrecededByAirBattles =
+        Properties.getRaidsMayBePreceededByAirBattles(data.getProperties());
+    for (final Unit bomber : bombers) {
+      final Set<Territory> bomberTargetTerritories = bomberMoveMap.get(bomber);
+      bomberTargetTerritories.removeIf(
+          terr ->
+              raidsMayBePrecededByAirBattles
+                  && AirBattle.territoryCouldPossiblyHaveAirBattleDefenders(
+                      terr, player, data, true));
+      determineBestBombingAttackForBomber(
+          attackMap, sortedUnitAttackOptions, bomberTargetTerritories, bomber);
+    }
+  }
+
+  /**
+   * Determines the best bombing attack territory for a single {@code bomber}.
+   *
+   * @param attackMap Attack map
+   * @param sortedUnitAttackOptions Sorted attack options
+   * @param bomberTargetTerritories Target territories
+   * @param bomber Bomber unit
+   */
+  private void determineBestBombingAttackForBomber(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Territory> bomberTargetTerritories,
+      final Unit bomber) {
+    final Predicate<Unit> bombingTargetMatch =
+        Matches.unitCanProduceUnitsAndCanBeDamaged()
+            .and(Matches.unitIsLegalBombingTargetBy(bomber));
+    Optional<Territory> maxBombingTerritory = Optional.empty();
+    int maxBombingScore = MIN_BOMBING_SCORE;
+    for (final Territory t : bomberTargetTerritories) {
+      final List<Unit> targetUnits = t.getUnitCollection().getMatches(bombingTargetMatch);
+      if (!targetUnits.isEmpty() && canAirSafelyLandAfterAttack(bomber, t)) {
+        final int noAaBombingDefense =
+            t.anyUnitsMatch(Matches.unitIsAaForBombingThisUnitOnly()) ? 0 : 1;
+        int maxDamageProduction = 0; // territory damage + target unit damage(s)
+        final TerritoryAttachment ta = TerritoryAttachment.get(t);
+        if (ta != null) {
+          maxDamageProduction = ta.getProduction();
+        }
+        int neededDamageUnits = 0;
+        final List<Unit> existingAttackingBombers = attackMap.get(t).getBombers();
+        for (Unit target : targetUnits) {
+          neededDamageUnits += target.getHowMuchDamageCanThisUnitTakeTotal(t);
+          int numExistingBombers = 0;
+          for (Unit existingBomber : existingAttackingBombers) {
+            if (Matches.unitCanProduceUnitsAndCanBeDamaged()
```

Comment:
Since we're iterating over `targetUnits`, we actually know that `Matches.unitCanProduceUnitsAndCanBeDamaged()` is already true since we checked it via `bombingTargetMatch`, so this can be simplified to just `(Matches.unitIsLegalBombingTargetBy(existingBomber)`.